.dagger/checks.go

```diff
@@ -2,151 +2,105 @@ package main
-	"golang.org/x/sync/errgroup"
+	"github.com/dagger/dagger/util/parallel"
 )
 
-// Check that everything works. Use this as CI entrypoint.
-func (dev *DaggerDev) Check(ctx context.Context,
-	// Directories to check
-	// +optional
-	targets []string,
-) error {
-	var routes checkRouter
-	routes.Add(Check{"docs", dag.Docs().Lint})
-	routes.Add(Check{"scripts/lint", dev.Scripts().Lint})
-	routes.Add(Check{"scripts/test", dev.Scripts().Test})
-	routes.Add(Check{"helm/lint", dag.Helm().Lint})
-	routes.Add(Check{"helm/test", dag.Helm().Test})
-	routes.Add(Check{"helm/test-publish", func(ctx context.Context) error {
-		return dag.Helm().Publish(ctx, "main", dagger.HelmPublishOpts{DryRun: true})
-	}})
-	routes.Add(dev.checksForSDK("sdk/go", dev.SDK().Go)...)
-	routes.Add(dev.checksForSDK("sdk/python", dev.SDK().Python)...)
-	routes.Add(dev.checksForSDK("sdk/typescript", dev.SDK().Typescript)...)
-	routes.Add(dev.checksForSDK("sdk/php", dev.SDK().PHP)...)
-	routes.Add(dev.checksForSDK("sdk/java", dev.SDK().Java)...)
-	routes.Add(dev.checksForSDK("sdk/rust", dev.SDK().Rust)...)
-	routes.Add(dev.checksForSDK("sdk/elixir", dev.SDK().Elixir)...)
-	routes.Add(dev.checksForSDK("sdk/dotnet", dev.SDK().Dotnet)...)
-
-	if len(targets) == 0 {
-		for route := range routes.children {
-			targets = append(targets, route)
-		}
-	}
-	eg := errgroup.Group{}
-	for _, check := range routes.Get(targets...) {
-		ctx, span := Tracer().Start(ctx, check.Name)
-		eg.Go(func() (rerr error) {
-			defer func() {
-				if rerr != nil {
-					span.SetStatus(codes.Error, rerr.Error())
-				}
-				span.End()
-			}()
-			return check.Check(ctx)
-		})
-	}
-	return eg.Wait()
+// Verify that generated code is up to date
+func (dev *DaggerDev) CheckGenerated(ctx context.Context) error {
+	_, err := dev.Generate(ctx, true)
+	return err
 }
 
-type Check struct {
-	Name  string
-	Check func(context.Context) error
+func (dev *DaggerDev) CheckReleaseDryRun(ctx context.Context) error {
+	return parallel.New().
+		WithJob("Helm chart", dag.Helm().CheckReleaseDryRun).
+		WithJob("CLI", dag.DaggerCli().CheckReleaseDryRun).
+		WithJob("Engine", dag.DaggerCli().CheckReleaseDryRun).
+		WithJob("SDKs", func(context.Context) error {
+			type dryRunner interface {
+				Name() string
+				CheckReleaseDryRun(context.Context) error
+			}
+			jobs := parallel.New()
+			for _, sdk := range allSDKs[dryRunner](dev) {
+				jobs = jobs.WithJob(sdk.Name(), sdk.CheckReleaseDryRun)
+			}
+			return jobs.Run(ctx)
+		}).
+		Run(ctx)
 }
 
-func (dev *DaggerDev) checksForSDK(name string, sdk sdkBase) []Check {
-	return []Check{
-		{
-			Name:  name + "/lint",
-			Check: sdk.Lint,
-		},
-		{
-			Name:  name + "/test",
-			Check: sdk.Test,
-		},
-		{
-			Name: name + "/test-publish",
-			Check: func(ctx context.Context) error {
-				return sdk.TestPublish(ctx, "HEAD")
-			},
-		},
-	}
+func (dev *DaggerDev) CheckLint(ctx context.Context) error {
+	return parallel.New().
+		WithJob("lint go packages", dev.CheckLintGo).
```

Comment:
nit: should `lint` be prefixed to all of this, given the function name already has `CheckLint`.

`CheckReleaseDryRun` doesn't have the same prefixing, just wandering if it's intentionally inconsistent here.